partners_api/partners_api_tests/taxi_engine_tests.cpp

```diff
@@ -0,0 +1,258 @@
+  // Only product id is important for us, all other fields are empty.
+  std::vector<taxi::Product> products1 =
+  {
+    {"1", "", "", "", ""},
+    {"2", "", "", "", ""},
+    {"3", "", "", "", ""},
+  };
+
+  std::vector<taxi::Product> products2 =
+  {
+    {"4", "", "", "", ""},
+    {"5", "", "", "", ""},
+    {"6", "", "", "", ""},
+  };
+
+  maker.Reset(reqId, 2, successCallback, errorNotPossibleCallback);
+  maker.ProcessProducts(reqId, taxi::Provider::Type::Uber, products1);
+  maker.ProcessProducts(reqId, taxi::Provider::Type::Yandex, products2);
+
+  TEST(providers.empty(), ());
+  TEST(errors.empty(), ());
+
+  maker.Reset(reqId, 3, successCallback, errorNotPossibleCallback);
+  maker.ProcessProducts(reqId, taxi::Provider::Type::Uber, products1);
+  maker.ProcessProducts(reqId, taxi::Provider::Type::Yandex, products2);
+  maker.DecrementRequestCount(reqId);
+  maker.MakeResult(reqId);
+
+  TEST_EQUAL(providers.size(), 2, ());
+  TEST_EQUAL(providers[0].GetType(), taxi::Provider::Type::Uber, ());
+  TEST_EQUAL(providers[1].GetType(), taxi::Provider::Type::Yandex, ());
+  TEST_EQUAL(providers[0][0].m_productId, "1", ());
+  TEST_EQUAL(providers[0][1].m_productId, "2", ());
+  TEST_EQUAL(providers[0][2].m_productId, "3", ());
+  TEST_EQUAL(providers[1][0].m_productId, "4", ());
+  TEST_EQUAL(providers[1][1].m_productId, "5", ());
+  TEST_EQUAL(providers[1][2].m_productId, "6", ());
+
+  maker.Reset(reqId, 2, successCallback, errorNotPossibleCallback);
+  maker.ProcessError(reqId, taxi::Provider::Type::Uber, taxi::ErrorCode::NoProducts);
+  maker.ProcessProducts(reqId, taxi::Provider::Type::Yandex, products2);
+  maker.MakeResult(reqId);
+
+  TEST_EQUAL(providers.size(), 1, ());
+  TEST_EQUAL(providers[0].GetType(), taxi::Provider::Type::Yandex, ());
+  TEST_EQUAL(providers[0][0].m_productId, "4", ());
+  TEST_EQUAL(providers[0][1].m_productId, "5", ());
+  TEST_EQUAL(providers[0][2].m_productId, "6", ());
+
+  maker.Reset(reqId, 2, successNotPossibleCallback, errorCallback);
+  maker.ProcessError(reqId, taxi::Provider::Type::Uber, taxi::ErrorCode::NoProducts);
+  maker.ProcessError(reqId, taxi::Provider::Type::Yandex, taxi::ErrorCode::RemoteError);
+  maker.MakeResult(reqId);
+
+  TEST_EQUAL(errors.size(), 2, ());
+  TEST_EQUAL(errors[0].m_type, taxi::Provider::Type::Uber, ());
+  TEST_EQUAL(errors[0].m_code, taxi::ErrorCode::NoProducts, ());
+  TEST_EQUAL(errors[1].m_type, taxi::Provider::Type::Yandex, ());
+  TEST_EQUAL(errors[1].m_code, taxi::ErrorCode::RemoteError, ());
+}
+
+UNIT_TEST(TaxiEngine_Smoke)
+{
+  // Used to synchronize access into GetAvailableProducts callback method.
+  std::mutex resultsMutex;
+  size_t reqId = 1;
+  taxi::ProvidersContainer providersContainer;
+  ms::LatLon const from(38.897724, -77.036531);
+  ms::LatLon const to(38.862416, -76.883316);
+
+  taxi::uber::SetUberUrlForTesting("http://localhost:34568/partners");
```

Comment:
Please, use UNIT_CLASS_TESTS, and in ctor set Uber URL for testing to localhost, and in dtor set it to the empty string.